python/pyspark/ml/tests/test_feature.py

```diff
@@ -357,6 +359,48 @@ def test_robust_scaler(self):
             self.assertEqual(str(model), str(model2))
             self.assertEqual(model2.getOutputCol(), "scaled")
 
+    def test_word2vec(self):
+        sent = ("a b " * 100 + "a c " * 10).split(" ")
+        df = self.spark.createDataFrame([(sent,), (sent,)], ["sentence"]).coalesce(1)
+
+        w2v = Word2Vec(vectorSize=3, seed=42, inputCol="sentence", outputCol="model")
+        w2v.setMaxIter(1)
+        self.assertEqual(w2v.getInputCol(), "sentence")
+        self.assertEqual(w2v.getOutputCol(), "model")
+        self.assertEqual(w2v.getVectorSize(), 3)
+        self.assertEqual(w2v.getSeed(), 42)
+        self.assertEqual(w2v.getMaxIter(), 1)
+
+        model = w2v.fit(df)
+        self.assertEqual(model.getVectors().columns, ["word", "vector"])
+        self.assertEqual(model.getVectors().count(), 3)
+
+        synonyms = model.findSynonyms("a", 2)
+        self.assertEqual(synonyms.columns, ["word", "similarity"])
+        self.assertEqual(synonyms.count(), 2)
+
+        # TODO(SPARK-50958): Support Word2VecModel.findSynonymsArray
```

Comment:
`Word2VecModel.findSynonymsArray` returns a complex type, cannot be automatically handled by literal, will try to fix it later